web/renderer/components/pageComponents/ConnectionsPage/NewConnection/context/utils.ts

```diff
@@ -4,7 +4,9 @@ import { ConfigState } from "./state";
 export function getConnectionUrl(state: ConfigState): string {
   if (state.connectionUrl) return state.connectionUrl;
   const prefix = state.type === DatabaseType.Mysql ? "mysql" : "postgresql";
-  return `${prefix}://${state.username}:${state.password}@${state.host}:${state.port}/${state.database}`;
+  return state.password
+    ? `${prefix}://${state.username}:${state.password}@${state.host}:${state.port}/${state.database}`
+    : `${prefix}://${state.username}@${state.host}:${state.port}/${state.database}`;
```

Comment:
Since this is kinda long and a lot is repeated it's not immediately clear what's different between the two. Option for better readability:
```suggestion
  const password = state.password ? `:${state.password}` : "";
  return `${prefix}://${state.username}${password}@${state.host}:${state.port}/${state.database}
```